### PR TITLE
Fix recording of multiple files.

### PR DESCRIPTION
**Pull request by Augustyniak**
Fix recording of multiple files on older devices
such as iPod 5th generation and iPhone 5C.

### CODE REVIEW

**Comment by Augustyniak** [comment]
@bartekchlebek ready to be reviewed 🎁

**Comment by Augustyniak** [comment]
Oh, wait. I missed something.

**Comment by Augustyniak** [comment]
Ready 🎁

File: TheAmazingAudioEngine/Utilities/AEUtilities.m [hunk 1]
```diff
@@ -53,86 +65,132 @@ BOOL AERateLimit(void) {
...
+        } else {
+            return kAudioFileWAVEType;
+        }
+    }
+}
+
+bool EXAudioStreamBasicDescription(AEAudioFileType fileType, double sampleRate, int channelCount, AudioStreamBasicDescription * asbd, NSError ** error) {
+
+    if (!asbd) {
+        return true;
```
Comment (bubski) [inline TheAmazingAudioEngine/Utilities/AEUtilities.m]: Now that I think about it, shouldn't we return `false`?

**Comment by bubski** [comment]
No more comments

File: TheAmazingAudioEngine/Utilities/AEUtilities.m (hunk 1 above)
Comment (Augustyniak) [inline TheAmazingAudioEngine/Utilities/AEUtilities.m]: To be honest I think that we should just crash here - calling this function with `absd` equal to `null` is a programmer mistake (as for me).

Comment (bubski) [inline TheAmazingAudioEngine/Utilities/AEUtilities.m]: I'm always pro crash on programmer error. 👍

Comment (Augustyniak) [inline TheAmazingAudioEngine/Utilities/AEUtilities.m]: If we don't want to crash then returning `true` makes sense for me. Everything went as user wanted.

Comment (Augustyniak) [inline TheAmazingAudioEngine/Utilities/AEUtilities.m]: Ok, I will crash here.

**Comment by Augustyniak** [comment]
@bartekchlebek fixed. Ready to be reviewed.

File: TheAmazingAudioEngine/Utilities/AEUtilities.m [hunk 2]
```diff
@@ -83,9 +83,7 @@ AudioFileTypeID EXAudioFileTypeToAudioFileTypeID(AEAudioFileType fileType) {
 
 bool EXAudioStreamBasicDescription(AEAudioFileType fileType, double sampleRate, int channelCount, AudioStreamBasicDescription * asbd, NSError ** error) {
 
-    if (!asbd) {
-        return true;
-    }
+    NSCAssert(asbd, @"Provide valid pointer to AudioStreamBasicDescription");
```
Comment (bubski) [inline TheAmazingAudioEngine/Utilities/AEUtilities.m]: Wouldn't we crash only on debug builds?

**Pull request merged**